src/test/FuzzerImpl.cpp

```diff
@@ -114,6 +115,104 @@ class FuzzTransactionFrame : public TransactionFrame
+{
+    VirtualClock clock;
+    mApp = createTestApplication(clock, getFuzzConfig(processID));
+
+    LedgerTxn ltx(mApp->getLedgerTxnRoot());
+
+    // setup the state, for this we only need to  pregenerate some accounts. For
+    // now we create 16 accounts, or enough to fill the first 4 bits such that
+    // we have a pregenerated account for the last 4 bits of the 32nd byte of a
+    // public key
+    for (int i = 0; i < 16; ++i)
+    {
+        PublicKey publicKey;
+        uint256 accountID;
+        accountID.at(31) = i;
+        publicKey.ed25519() = accountID;
+
+        // manually construct ledger entries, "creating" each account
+        LedgerEntry newAccountEntry;
+        newAccountEntry.data.type(ACCOUNT);
+        auto& newAccount = newAccountEntry.data.account();
+        newAccount.thresholds[0] = 1;
+        newAccount.accountID = publicKey;
+        newAccount.seqNum = 0;
+        // to create "interesting" balances we utilize powers of 2
+        newAccount.balance = pow(2, i);
+        ltx.create(newAccountEntry);
+
+        // select the first pregenerated account to be the hard coded source
+        // account for all transactions
+        if (i == 0)
+            mSourceAccountID = publicKey;
+    }
+
+    // commit these to the ledger so that we have a starting, persistent state
+    // to fuzz test against -- should be the only stateful/effectful action
+    ltx.commit();
+}
+
+void
+TransactionFuzzer::inject(stellar::XDRInputFileStream& in)
+{
+    // for tryRead, in case of fuzzer creating an ill-formed xdr, generate an
+    // xdr that will trigger a non-execution path so that the fuzzer realizes it
+    // has hit an uninteresting case
+    auto tryRead = [](stellar::XDRInputFileStream& in,
+                      xdr::xvector<Operation>& m) {
+        try
+        {
+            return in.readOne(m);
+        }
+        catch (...)
+        {
+            m.clear(); // we ignore transactions with 0 operations
+            return true;
+        }
+    };
+
+    xdr::xvector<Operation> ops;
+    while (tryRead(in, ops))
+    {
+        // direct the fuzzer such that it will "learn" that five transactions
+        // is the maximum since nothing over five transactions will ever
+        // result in newly discovered execution paths
+        if (ops.size() == 0 || ops.size() > 5)
+            return;
+
+        // construct transaction
+        auto txFramePtr = ::FuzzTransactionFrame::create(mSourceAccountID, ops,
+                                                         mApp->getNetworkID());
+
+        try
+        {
+            // attempt to apply transaction
+            LedgerTxn ltx(mApp->getLedgerTxnRoot());
+            txFramePtr->attemptApplication(*mApp, ltx);
+
+            // we need an extra db cache clear here since the db prepared
+            // statement cache may persist between tryRead loops
+            mApp->getDatabase().clearPreparedStatementCache();
```

Comment:
Why is this needed? You are already clearing this in `attemptApplication`.

Reply:
I cannot quite figure out why it is needed -- for some reason, calling this before the `XDRStream` reads another `XDR` in `tryRead` makes the stability go from `91% --> 95%` even though I am fairly confident the two are wholly unrelated.

Reply:
*Another general observation:* turns out that the first `clearPreparedStatementCache` has generally no affect on stability, given the existence of this, the second `clearPreparedStatementCache` call. This makes sense given that it clears the cache after the `tx attempted apply`, before another loop iteration, and before `XDRStream`.

Reply:
`resetTxInternalState` should call `clearPreparedStatementCache`

Reply:
I suspect you need to call `clearPreparedStatementCache` after `ltx` goes out of scope just to be sure things are reset the way you want